lib/waterline/collection/index.js

```diff
@@ -54,3 +54,42 @@ inherits(Collection, Query);
 
 // Make Extendable
 Collection.extend = extend;
+
+
+/**
+* Resolve Collection
+*
+* Helper to resolve `tenant` into
+* Tenant-specific Collection, if applicable
+*
+* @param {String} connection name
+* @param {Function} callback(err, resolvedCollection)
+* @return {WLCollection} this
+*/
+Collection.prototype._resolveCollection = function(connectionName, cb) {
+    // Get Connection
+    var connectionName = connectionName || this.adapter.dictionary.identity;
+    var connection = this.connections[connectionName];
+    var config = connection.config;
+    // Check if Multi-Tenant enabled
+    if (config.isMultiTenant) {
+        // Is Multi-Tenant connection
+        // Check if there is a Tenant set
+        var tenant = config.tenant;
+        if (tenant) {
+            // Tenant is set
+            // Use Tenant-specific collection
+            // Check if Tenant's connection has been loaded
+            //   console.log('select tenant', config.tenant);
+            this.tenant(tenant, function(err, tenantCollection) {
+                return cb(null, tenantCollection);
```

Comment:
```js
return cb(null, tenantCollection);
```

Should be

```js
return cb(err, tenantCollection);
```

Note the missing `err` when calling the callback.

Special thanks to @chipgata for being a tester and showing some strange and unexpected errors. Fixing this should clear things up.